Configs/.local/bin/Hyprdots

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+source ~/.config/hypr/scripts/globalcontrol.sh
+HyprDir="$HOME/.config/hypr/scripts"
+ScriptDir="$CloneDir/Scripts"
+func="$@"
+
+params() {
+  func=$1
+if awk "/^$func\(\) \{/,/}/" $0 | grep > /dev/null  'case'; then
+  awk "/^$func\(\) \{/,/}/" $0 | grep -E '^\s+\w+\)' | awk 'BEGIN { FS = "[()#?]" } { sub(/^[ \t]+/, "", $1); printf "                          %-15s %s\n","["$1"]", $4 }'
+fi
+}
+
+guide() { #? Show longer help message
+  echo "Usage: Hyprdots [function] [parameters]"
+  echo ""
+  grep -E '^\w+\(\) \{ #\?' $0 | while read -r line; do
+    func=$(echo "$line" | awk 'BEGIN { FS = "[()#?]" } { print $1 }')
+    desc=$(echo "$line" | awk 'BEGIN { FS = "[()#?]" } { print $5 }')
+printf "  %-20s %s\n" "[$func]" "*$desc"
+      params "$func"
+       done
+}
+
+restore() { #? Restore Hyprdots Conigurations
+./install.sh  -r
+  }
+
+update() { #? Pull Updates From Hyprdots Repository 
+git stash 2> /dev/null ; git pull ; git stash pop 2> /dev/null 
+  }
+
+install() { #? Fresh install your config also installs apps. 
+./install.sh
+  }
+
+theme() { #? Theme Related Actions
+  # Function for managing themes
+case $1 in 
+        set) #? Set Theme
+            $HyprDir/themeswitch.sh -s
+        ;;
+        next) #? Next Theme
+            $HyprDir/themeswitch.sh -n
+        ;; 
+        prev) #? Previous Theme
+            $HyprDir/themeswitch.sh -p
+        ;;
+        select) #? Selec
+            $HyprDir/themeselect.sh
+        ;;
+        patch) #? Patch A Theme from different repository.
+            shift
+            cd $ScriptDir
+            ./themepatcher.sh "$@"
+        ;;
+        *)
+         echo "$(params "${func}" )"
+        ;;
+esac
+}
+
+wallpaper() { #? Wallpaper related Functions
+  # Function for managing wallpapers
+  echo "Setting wallpaper with parameters: $@"
+  case $1 in 
+        set)
+            shift
+            if [ -z "$@" ]; then echo "Missing: /Path/to/Wallpaper" ; exit 1 ; fi
+            $HyprDir/swwwallpaper.sh -s "$@"
+        ;;
+        next)
+            $HyprDir/swwwallpaper.sh -n 
+        ;;
+        prev)
+            $HyprDir/sswwwallpaper.sh -p
+        ;;
+        select)
+            $HyprDir/swwwallselect.sh
+        ;;
+        *)
+        echo "$(params "${func}" )"
+        ;;
+esac
+}
+
+sddm() { #? Sddm Things
+  # Function for managing SDDM settings
+  echo "Setting SDDM with parameters: $@"
+  case $1 in 
+        set)
+            $HyprDir/sddmwall.sh
+        ;;
+esac
+}
+wallbash(){ #? Wallbash
+  case $1 in 
+        toggle)
+            $HyprDir/wallbashtoggle.sh
+        ;;
+esac
+}
+
+waybar() { #? Waybar related Things
+    echo $1
+case $1 in
+        edit)
+   # Edit the configuration file
+   code ~/.config/waybar/config.ctl
+   # Restart Waybar when the configuration file changes
+        inotifywait -e modify $HOME/.config/waybar/config.ctl
+    notify-send  "Changes On waybar"
+ $HOME/.config/hypr/scripts/themeReload.sh all
```

Comment:
themereload? are you working on this?

Reply:
Oh it just runs the reload part on the themeswitch script I guess?  I just use this for testing. Can I know which part of the script I can use to call for reloading? 
Also i guess this script will be ready if I can learn to use this inotify-tools package as it will make the changing of waybar control slightly dynamic. Especially the vscode extension setting too.